torch/_dynamo/variables/torch.py

```diff
@@ -272,6 +293,7 @@ def call_function(
         constant_args = check_constant_args(args, kwargs)
         unspec_python_args = check_unspec_python_args(args, kwargs)
         options = VariableTracker.propagate(self, args, kwargs.values())
+        check_allowed_op(self.value)
```

Comment:
One thing I'm a bit confused about:

(1) why do `check_allowed_op` and `check_allowed_overload_packet` get run in different places?

(2) It seems like `check_allowed_op` also has special handling for when it sees an OverloadPacket that only contains 1 overload? Why isn't that already handled by `check_allowed_overload_packet`?

Reply:
TL;DR: I'm doing some defensive programming (one of these callsites is much more robust than the other, so we prefer to put as much as we can there)

> (1) why do check_allowed_op and check_allowed_overload_packet get run in different places?

check_allowed_op runs at the beginning of TorchVariable.call_function.
check_allowed_overload_packet runs somewhere in the middle of TorchVariable.call_function.

They run at different places because:
- The beginning of TorchVariable.call_function is a robust place to put the check: if the TorchVariable is an opoverload, we check the tag and graph break accordingly.
- check_allowed_overload_packet requires the sample inputs to do overload resolution. So the check needs to be somewhere in the middle of TorchVariable.call_function where we can actually acquire the sample inputs.
- The middle of TorchVariable.call_function is not the most robust place because there is a large chain of if-then-else. I'm worried someone will come along and add a special case for OpOverloadPacket that bypasses the logic. There are already special cases for things like torch.ops.aten.sym_sizes in the middle that bypass this logic (but we already tag all aten ops with the pt2_compliant_tag, so these special cases are OK).

> (2) It seems like check_allowed_op also has special handling for when it sees an OverloadPacket that only contains 1 overload? Why isn't that already handled by check_allowed_overload_packet?

Because most custom ops only have 1 overload, we can handle this case without actually grabbing the sample inputs. check_allowed_op is the more robust check due to where it is located in the code, so I put the check for 1 overload there.

Reply:
Something more robust is if we create a new TorchOpsVariable to represent everything in `torch.ops.*` and then refactor all the existing code to handle that (to go into TorchOpsVariable instead of TorchVariable). This makes the if-then-else a bit less brittle